### PR TITLE
Allow general user to update their own tuit

### LINKED ISSUE #12: Update Tuit React component for admin and general user view and RUD

### PR DESCRIPTION
**Pull request by averywpx**

### CODE REVIEW

File: src/components/tuits/tuit.js [hunk 1]
```diff
@@ -47,14 +50,42 @@ const Tuit = ({tuit, deleteTuit, likeTuit, dislikeTuit}) => {
                     <i onClick={() => deleteTuit(tuit._id)}
                        className="fas fa-remove tuit-button fa-2x fa-pull-right"/>
```
Comment (aliu19) [inline src/components/tuits/tuit.js:51]: Update to have the pencil, green checkmark, and red trash can as shown in main branch (may be fixed during selection during merge?)

**Review by aliu19 — CHANGES_REQUESTED** [review]

**Review by aliu19 — APPROVED** [review]
Update to have the pencil, green checkmark, and red trash can in the tuit as shown in main branch, which should be fixed from resolving merge conflicts

**Comment by caijiah** [comment]
The code looks fine, but I don't remember we want general users to edit tuits.

File: src/components/tuits/tuit.js [hunk 2]
```diff
@@ -47,14 +50,42 @@ const Tuit = ({tuit, deleteTuit, likeTuit, dislikeTuit}) => {
...
                 }
-                <Link to={`/tuit/${tuit._id}`}>
+                {/* <Link to={`/tuit/${tuit._id}`}>
                     <i className="float-end tuit-button fas fa-circle-ellipsis me-1"/>
-                </Link>
+                </Link> */}
+                {
+                    !editing &&
+                    <i className='float-end fa fa-cog edit-button'
```
Comment (caijiah) [inline src/components/tuits/tuit.js:58]: Need change to pen icon.

File: src/components/tuits/tuit.js [hunk 3]
```diff
@@ -47,14 +50,42 @@ const Tuit = ({tuit, deleteTuit, likeTuit, dislikeTuit}) => {
...
+                       }}/>
+                }
+                {
+                    editing &&
+                    <div className='up-del-buttons'>
+                        <i onClick={() => {
+                            setEditing(false)
+                            updateTuit(tuitCache) 
+                        }}
+                        className="float-end fa fa-check"/>
```
Comment (caijiah) [inline src/components/tuits/tuit.js:70]: You probably also need delete button here.

**Review by caijiah — CHANGES_REQUESTED** [review]
We have discussed in chat that we want only owner of the tuit can edit the tuit.

**Review by caijiah — APPROVED** [review]

**Pull request closed without merging**